vignettes/algorithm_data.Rmd

```diff
@@ -0,0 +1,151 @@
+---
+title: "Description of data objects related to the algorithm"
+output: rmarkdown::html_vignette
+bibliography: references.bib
+csl: vancouver.csl
+vignette: >
+  %\VignetteIndexEntry{Design}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+library(dplyr)
+```
+
+## Contents
+
+This document describes the structure of the data components processed
+by the algorithm.
+
+## Data components
+
+The algorithm uses five different types of data, contained in five
+register sources:
+
+1.  Hospital diagnoses
+    -   The National Patient Register [Landspatientregisteret]
+2.  Prescription drugs purchased
+    -   The Register of Pharmaceutical Sales
+        [Lægemiddelstatistikregisteret]
+3.  Hemoglobin-A1c tests
+    -   The Register of Laboratory Results for Research
+        [Laboratoriedatabasens Forskertabel]
+4.  Diabetes-specific podiatrist services
+    -   The National Health Insurance Service Register
+        [Sygesikringsregisteret]
+5.  Sex & date of birth
+    -   The Danish Civil Registration System [CPR-registeret]
+
+In a future revision, the algorithm can also utilise the Danish Medical
+Birth Register to extend the period of time of valid inclusions further
+back in time compared to what is possible using obstetric codes from the
+National Patient Register.
+
+## Pre-processing steps
+
+This section describes the necessary steps required to format raw data
+into a format that can be fed as input to the algorithm. The description
+assumes that raw data is stored/structured in the most common format for
+raw data provided on Statistics Denmark's servers (from our experience).
+
+Using the most common scenario when working with the above data on
+Statistics Denmark's servers, this paragraph lists the common register
+abbreviations/raw file names, their structure (year-on-year files vs. a
+large single file, plus changes/breaks over time), raw variable names
+and relevant values. Variable names are presented in lower case here,
+but case may vary between data sources (and even between years in the
+same data source) in real data.
+
+Depending on the contents and format of your specific raw data, you may
+need to adapt the pre-processing pipeline accordingly.
+
+## Structure of raw data
+
+### National Patient Register
+
+The National Patient Register contains several tables and types of data.
+The algorithm uses only hospital diagnosis data, which is contained in
+two tables:
+
+1.  A table containing administrative information, e.g. personal ID,
+    `pnr`/`cpr`, and the first date of the contact,
+    `d_inddto`/`dato_start`.
+
+    -   Named `lpr_adm` in the LPR2-formatted data prior to 2019, and
+        `kontakter` in contact-based LPR3-formatted data from 2019
```

Comment:
Do you want to use the English abbreviation `DNPR`?

Reply:
Maybe also add what the abbreviation comes from? I.e., Landspatientregisteret? 
Or do we assume readers know this?